### ISSUE TITLE
Place nodes missing if there is nearby place node with identical name (improve filtering heuristic)

### ISSUE THREAD

**Issue by Timmy-Tesseract** [organicmaps/organicmaps#7824]
**Describe the issue**
OM appears to filter out certain place nodes during map generation if there is a nearby place node with identical name. Affected place nodes will not be rendered and will not show up in search results. This happens despite place nodes having a number of attributes clearly marking them as distinct entities, in some cases even if value of place tag is different.

**Example 1**
2x place=village
distance between place nodes 2.1 km
located in different admin_level=6 entities
https://www.openstreetmap.org/node/6917101102 (exists in OM)
https://www.openstreetmap.org/node/6910742180 (missing in OM)

![Bagumbayan](https://github.com/organicmaps/organicmaps/assets/71176747/22866937-0a32-485c-9419-9c8f64ae500c)

**Example 2**
place=village and place=hamlet
distance between place nodes 848 m
located in different admin_level=4 entities
https://www.openstreetmap.org/node/5249604308 (exists in OM)
https://www.openstreetmap.org/node/6787296079 (missing in OM)

![Tibungol](https://github.com/organicmaps/organicmaps/assets/71176747/fdb30874-13b7-407e-818f-76174e6477a6)

**Example 3**
2x place village
distance between place nodes 2.4 km
located in different admin_level=3 entities
https://www.openstreetmap.org/node/4462800161 (exists in OM)
https://www.openstreetmap.org/node/8791290763 (missing in OM)

![Awao](https://github.com/organicmaps/organicmaps/assets/71176747/f7cb7069-0b1b-4e96-a9a9-97416f85ac3b)

**Limits to reproducibility**
- Issue only seems to occur if proximity is under a certain threshold. For above mentioned example 1 there is a [third village](https://www.openstreetmap.org/node/8809163222) with identical name 17 km away which is not ommited in OM.
- Issue appears to not affect all close proximity place nodes with identical names. In OM screenshot for example 1 there are 2 place=neighbourhood with name "Purok 4" displayed correctly despite distance of only 820 m which is less than the missing place nodes in all three examples.

**Comment by biodranik** [comment]
@Timmy-Tesseract what does it mean physically, _two different villages with the same name_?

**Comment by Timmy-Tesseract** [comment]
Yes there are two distinct settlements both with their own government offices, schools, health facilities, etc.

Historical background is usually that a sparsely populated area was already named before administrative boundaries were delineated. Then once territorry was divided by a municipal boundary both municipalities established their own village using the already existing name.

**Comment by pastk** [comment]
FYI @vng

**Comment by vng** [comment]
Yes, we do filter similar neighbor OSM places because of often mapping duplicates (Node, Way, Relation, and sometimes several of them) that _logically_ define one place.

What can we do here? Probably enhance the _similarity_ heuristic or add exceptions somehow ..

See generator/place_processor.cpp, the logic is very _heuristic_ there :)

**Comment by biodranik** [comment]
@vng is it possible to generate a dump of all such close, filtered features from our generator, to manually take a look on them, and maybe add exceptions based on their ids?

**Comment by muralito** [comment]
> **Example 3** 2x place village distance between place nodes 2.4 km located in different admin_level=3 entities https://www.openstreetmap.org/node/4462800161 (exists in OM) https://www.openstreetmap.org/node/8791290763 (missing in OM)

This two nodes has different value in tag is_in:municipality. This data could be used to feed the heuristic, and also to display it in the text list result, so the user can differentiate between them.

**Comment by Timmy-Tesseract** [comment]
Issue has been resolved with the latest update. Thank you for fixing this.
